pkg/cmd/backport/main.go

```diff
@@ -0,0 +1,523 @@
+// Copyright 2018 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/cockroachdb/cockroach/pkg/util/executil"
+	"github.com/google/go-github/github"
+	"github.com/pkg/errors"
+	"github.com/spf13/pflag"
+	"golang.org/x/oauth2"
+)
+
+const usage = `usage: backport [-c <commit>] [-r <release>] <pull-request>...
+   or: backport [--continue|--abort]`
+
+const helpString = `backport attempts to automatically backport GitHub pull requests to a
+release branch.
+
+By default, backport will cherry-pick all commits in the specified PRs.
+If you explicitly list commits on the command line, backport will
+cherry-pick only the mentioned commits.
+
+If manual conflict resolution is required, backport will quit so you
+can use standard Git commands to resolve the conflict. After you have
+resolved the conflict, resume backporting with 'backport --continue'.
+To give up instead, run 'backport --abort'.
+
+To determine what Git remote to push to, backport looks at the value of
+the cockroach.remote Git config option. You can set this option by
+running 'git config cockroach.remote REMOTE-NAME'.
+
+Options:
+
+      --continue           resume an in-progress backport
+      --abort              cancel an in-progress backport
+  -c, --commit <commit>    only cherry-pick the mentioned commits
+  -r, --release <release>  select release to backport to
+      --help               display this help
+
+Example invocations:
+
+    $ backport 23437
+    $ backport 23389 23437 -r 1.1 -c a26506b
```

Comment:
Wait, so you're choosing two PRs but filtering out a single commit? Is that a "I know this SHA but not sure which of the two PRs it's in so I'm just gonna list them both"?

Reply:
Oops, dumb example. It was supposed to show commit filtering and multiple PRs all at once. Updated to:

```
backport 23389 23437 -r 1.1 -c 00c6a87 -c a26506b
```